internal/apicodec/codec_v2.go

```diff
@@ -639,30 +618,50 @@ func (c *codecV2) encodeRange(start, end []byte, reverse bool) ([]byte, []byte)
 	}
 	var encodedEnd []byte
 	if len(end) > 0 {
-		encodedEnd = c.encodeKey(end)
+		encodedEnd = c.EncodeKey(end)
 	} else {
 		encodedEnd = c.endKey
 	}
-	return c.encodeKey(start), encodedEnd
+	return c.EncodeKey(start), encodedEnd
 }
 
-// decodeRange maps encodedStart and end back to normal start and
+// DecodeRange maps encodedStart and end back to normal start and
 // end without APIv2 prefixes.
-// Note that it uses empty byte slice to mark encoded start/end
-// that lies outside the prefix's range.
-func (c *codecV2) decodeRange(encodedStart, encodedEnd []byte) ([]byte, []byte) {
+func (c *codecV2) DecodeRange(encodedStart, encodedEnd []byte) ([]byte, []byte, error) {
 	var start, end []byte
-	if bytes.Compare(encodedStart, c.prefix) < 0 {
-		start = []byte{}
-	} else {
-		start = encodedStart[len(c.prefix):]
+	var err error
+	start, err = c.decodeKey(encodedStart)
+	if err != nil {
+		return nil, nil, err
 	}
-	if len(encodedEnd) == 0 || bytes.Compare(encodedEnd, c.endKey) >= 0 {
-		end = []byte{}
-	} else {
-		end = encodedEnd[len(c.endKey):]
+	end, err = c.decodeKey(encodedEnd)
+	if err != nil {
+		return nil, nil, err
 	}
-	return start, end
+	return start, end, nil
+}
+
+func (c *codecV2) EncodeKey(key []byte) []byte {
+	return append(c.prefix, key...)
+}
+
+func (c *codecV2) DecodeKey(key []byte) ([]byte, []byte, error) {
```

Comment:
Sorry about the interface change, the latest `api-v2` branch has changed it back to return only the user key and the error.